### PR TITLE
CoopFramework Synchronization Trigger Change bugfix (Based on Automatic Sync Initialization PR)

### LINKED ISSUE #256: CoopFramework Synchronization Trigger Change
Currently syncronization broadcast is called on skipped functions. Change this so that sync is only broadcasted when the method is called.

See 
CallBehaviourBuilder Broadcast method
CoopManaged Dispatch method

### PR DESCRIPTION
**Pull request by WakooMan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] All new classes have class-level documentation comments, if there are any at all
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## What is the current behaviour?

- When the function is skipped the call is still broadcasted.

## What is the new behaviour?

- When the function is skipped, the call is not broadcasted further.

## Question For Future Problems Probably
- In Dispatch method if one behaviour skips the method, but the list has a behaviour that doesn't skip the method, than should the method be broadcasted?
## Resolves

- CoopFramework Synchronization Trigger Change issue.

## Other information

### CODE REVIEW

File: source/CoopFramework/CoopManaged.cs [hunk 1]
```diff
@@ -260,20 +260,25 @@ private static void OnAutoConstructed(TSelf newInstance)
             foreach (var behaviour in behaviours)
             {
                 if (!behaviour.AppliesTo(eOriginator, instanceResolved)) continue;
-                if (behaviour.DoBroadcast)
+                ECallPropagation prop = ECallPropagation.CallOriginal;
+                if (behaviour.MethodCallHandler != null)
+                { 
+                    prop = behaviour.MethodCallHandler.Invoke(new PendingMethodCall(invokableId,
```
Comment (garrettluskey) [inline source/CoopFramework/CoopManaged.cs]:
```suggestion
                    prop = behaviour.MethodCallHandler?.Invoke(new PendingMethodCall(invokableId,
```

File: source/CoopFramework/CoopManaged.cs [hunk 2]
```diff
@@ -260,20 +260,25 @@ private static void OnAutoConstructed(TSelf newInstance)
...
+                        behaviour.SynchronizationFactory,
+                        instanceResolved,
+                        args));
```
Comment (garrettluskey) [inline source/CoopFramework/CoopManaged.cs]:
```suggestion
                        args)) ?? prop; 
```

**Review by garrettluskey — CHANGES_REQUESTED** [review]

File: source/CoopFramework/CoopManaged.cs (hunk 2 above)
Comment (WakooMan) [reply source/CoopFramework/CoopManaged.cs]: Done

File: source/CoopFramework/CoopManaged.cs (hunk 1 above)
Comment (WakooMan) [reply source/CoopFramework/CoopManaged.cs]: Done

**Comment by garrettluskey** [comment]
Refactoring entire sync framework. Changes no longer required

**Pull request closed without merging**